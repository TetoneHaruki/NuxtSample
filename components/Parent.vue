<script setup lang="ts">
import Child from './Child.vue';
</script>
<template>
    <div>
        <h1 class="parent">Parent</h1>
        <Child />
    </div>
</template>