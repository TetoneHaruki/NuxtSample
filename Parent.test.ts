import { describe, expect, it } from "vitest";
import { mount, shallowMount } from "@vue/test-utils";
import Parent from "./components/Parent.vue";

describe("Parent", () => {
    it("mount", () => {
        const component = mount(Parent);
        expect(component.element).toMatchSnapshot();
    });
    it('shallowMount', () => {
        const component = shallowMount(Parent);
        expect(component.element).toMatchSnapshot();
    });
});
