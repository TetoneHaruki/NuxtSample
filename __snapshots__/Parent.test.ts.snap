// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Parent > mount 1`] = `
<div>
  <h1
    class="parent"
  >
    Parent
  </h1>
  <div>
    <h1>
      Child
    </h1>
  </div>
</div>
`;

exports[`Parent > shallowMount 1`] = `
<div>
  <h1
    class="parent"
  >
    Parent
  </h1>
  <child-stub />
</div>
`;
